akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -670,9 +670,12 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
-    flushing.recover { case _ ⇒ Done }.map { _ ⇒
-      killSwitch.shutdown()
 
+    for {
+      _ ← flushing.recover { case _ ⇒ Done }
+      _ = killSwitch.shutdown()
+      _ ← streamsCompleted
```

Comment:
True. Do we really want to silence any errors though? Collect and log or something like that?